File: Language/Java/OOPS/Abstraction.java
//Abstraction
import data.*;
interface I2 {
    public void print();
}
class C5 extends AC {
    public void display() {
        System.out.println(this.id);
    }
}
class C6 implements I1, I2 {
    public void display() {
        System.out.println(amt);
    }
    public void print() {
        System.out.println("Hello Java");
    }
}

public class Abstraction {
    public static void main(String[] args) {
        C5 obj1=new C5();
        C6 obj2=new C6();

        System.out.println(obj1.getId());
        obj1.display();

        System.out.println(obj2.getAmt());
        obj2.display();
        obj2.print();
    }
}

File: Language/C++/OOPS/Abstraction.cpp
//Abstraction
#include<iostream>
using namespace std;
class C1 {
    public:
        virtual void print() = 0;
};
class C2: public C1 {
    public:
        void print() {
                cout<<"Hello C"<<endl;
            }
};
class C3: public C1 {
    public:
        void print() {
                cout<<"Hello C++"<<endl;
            }
};
class C4: public C2, public C3 {};
class C5 {
    public:
        void display() {
            cout<<"Hello C#"<<endl;
        }
};
class C6: virtual public C5 {};
class C7: virtual public C5 {};
class C8: public C6, public C7 {
    public:
        C8(): C6(), C7(), C5() {}
};

int main()
{
    C4 obj1;
    obj1.C2::print();
    obj1.C3::print();

    C8 obj2;
    obj2.display();
    return 0;
}